Homework 1 for Machine learning with python <br>
Data source:https://www.kaggle.com/aishwaryamuthukumar/cars-dataset-audi-bmw-ford-hyundai-skoda-vw <br>
1. Choosen dataset: toyota.csv

Import required Libraries

In [62]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

import pandas as pd
import numpy as np

2. Load data

In [63]:
import os

#Changing working directory
os.chdir(r'H:\EMK_Data analysis with python')

#Read 'csv' file
df = pd.read_csv('toyota.csv',encoding = 'unicode_escape')

display(df.head(10))
print(df.shape)

model  year  price transmission  mileage fuelType   mpg  engineSize
0   GT86  2016  16000       Manual    24089   Petrol  36.2         2.0
1   GT86  2017  15995       Manual    18615   Petrol  36.2         2.0
2   GT86  2015  13998       Manual    27469   Petrol  36.2         2.0
3   GT86  2017  18998       Manual    14736   Petrol  36.2         2.0
4   GT86  2017  17498       Manual    36284   Petrol  36.2         2.0
5   GT86  2017  15998       Manual    26919   Petrol  36.2         2.0
6   GT86  2017  18522       Manual    10456   Petrol  36.2         2.0
7   GT86  2017  18995       Manual    12340   Petrol  36.2         2.0
8   GT86  2020  27998       Manual      516   Petrol  33.2         2.0
9   GT86  2016  13990       Manual    37999   Petrol  36.2         2.0

(6738, 8)


In [64]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6738 entries, 0 to 6737
Data columns (total 8 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   model         6738 non-null   object 
 1   year          6738 non-null   int64  
 2   price         6738 non-null   int64  
 3   transmission  6738 non-null   object 
 4   mileage       6738 non-null   int64  
 5   fuelType      6738 non-null   object 
 6   mpg           6738 non-null   float64
 7   engineSize    6738 non-null   float64
dtypes: float64(2), int64(3), object(3)
memory usage: 421.2+ KB


In [65]:
print(df.columns)

Index(['model', 'year', 'price', 'transmission', 'mileage', 'fuelType', 'mpg',
       'engineSize'],
      dtype='object')


(B) Separating the numeric features and target variable

In [66]:
#X= Feature variable/Independed variable, Y= Target/ Depended variable

features = ['year','mileage', 'mpg', 'engineSize']
target = ['price']

X = df[features]
y = df[target]

print(X.shape, y.shape)

(6738, 4) (6738, 1)


Here, (6738, 4) = (Row , Column) in X dataframe <br>
(6738, 1) = (Row , Column) in y dataframe <br>

'model', 'transmission' and 'fuelType' columns are categorical data.


(C) Split the original dataset into the train set (80%) and the test set (20%), 


In [67]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=1)

print(X_train.shape, X_test.shape, y_train.shape, y_test.shape)

(5390, 4) (1348, 4) (5390, 1) (1348, 1)


Data set (6738, 4) (6738, 1) are divided into 
train set=(5390, 4) and test set = (1348, 4) for X <br> 
train set=(5390, 1) and test set = (1348, 1) for y

(D) Perform Linear Regression and Predict the 'Price' from the test set, 


In [68]:
#LinearRegression
model = LinearRegression()
model = model.fit(X_train, y_train)

In [69]:
#Prediction
y_pred = model.predict(X_test)

print(y_pred)


[[17047.903718  ]
 [10756.12766572]
 [ 8994.28521115]
 ...
 [11317.2571353 ]
 [ 8435.95079987]
 [ 9205.44764864]]


Predicted y data based on X test value

In [70]:
print(y_test)

      price
6334  13998
2256   9420
4429   7901
2353  14810
5903  20365
...     ...
559   22362
1941  13200
2043   9995
3730   7998
1266   7688

[1348 rows x 1 columns]


Actual y test data based on X test value

(E) Find the RMSE value from the actual test data and the predicted data.

RMSE= Root_Mean_Suqared_Error <br>
least value is preferable (close to zero)

In [71]:
#Finding error between actual and predicted data 
RMSE = mean_squared_error(y_test, y_pred, squared=False)
print(RMSE)

2916.148404254022


The model gives overall error 2916.14 between test and predicted data. 